test/framework/src/main/java/org/elasticsearch/transport/nio/SocketEventHandler.java

```diff
@@ -78,8 +79,8 @@ public void handleConnect(NioSocketChannel channel) {
      * @param channel that was connecting
      * @param exception that occurred
      */
-    public void connectException(NioSocketChannel channel, Exception exception) {
-        logger.trace("failed to connect to channel", exception);
+    void connectException(NioSocketChannel channel, Exception exception) {
+        logger.debug(new ParameterizedMessage("failed to connect to socket channel: {}", channel), exception);
```

Comment:
Since this is at the debug level, should you use a supplier here to avoid instantiating the parameterized message if not needed?